### ISSUE TITLE
Surge (1.7.1) crashes in Bitwig Studio (3.2.8) [Demo Version] (Arch Linux)

### ISSUE THREAD

**Issue by jc-SpaceXp** [surge-synthesizer/surge#2726]
Successfully installed surge via: https://www.archlinux.org/packages/community/x86_64/surge/
i.e. `pacman -S surge`. Installed Bitwig Studio via the AUR and all plugins except for Surge work. Although Surge is the only VST3 plugin I have installed.

Upon opening the plugin (32-bit or 64-bit) doing most actions immediately crash the plugin. Examples [while Surge is the active window]: closing the window, pressing shift or caps-lock (used to play midi notes in Bitwig), pressing any lowercase key and alt-tabbing. Note: audio does come out of surge. I can left click and right click in Surge but I'm unable to fine-control any parameters.

Here is a partial output of engine.log from Bitwig:
```
About to start the following process:  /opt/bitwig-studio/bin/BitwigPluginHost64 host Surge-Synth-Team-Surge-6 44100 512 3564-6 4 not-dpi-aware 6 pulseaudio pulseaudio
Creating plugin audio thread proxy 0
PluginHost: Starting new audio thread for processing plugin 0
Error setting realtime priority for thread to -1
PluginHost: Loading initial plugin state: /home/USERNAME/.BitwigStudio/plugin-states/2395799b-01d4-4221-a19a-bc4bfaa7e814.vstpreset
Applying plugin IO state
Applying plugin instance state
Sending a shutdown request to the plugin host
PluginHost: Destroying plugin with id 1
PluginHost: Shutting down...
PluginHost: Shutting down request reader thread
PluginHost: Request reader thread finished
PluginHost: Shutting down audio threads
PluginHost: Exiting due to shutdown
PluginHost: exiting with code 0
Could not read async reply: End of stream
Waiting for plugin host process to exit
Plugin host process exited with code: 0
Waiting for plugin host process to exit
Plugin host process exited with code: 0
About to start the following process:  /opt/bitwig-studio/bin/BitwigPluginHost64 host Surge-Synth-Team-Surge-7 44100 512 3564-7 4 not-dpi-aware 6 pulseaudio pulseaudio
Creating plugin audio thread proxy 0
PluginHost: Starting new audio thread for processing plugin 0
Error setting realtime priority for thread to -1
Applying plugin IO state
Applying plugin instance state
IPlugFrame is a runloop 0x2052f80
Could not read async reply: End of stream
Killing pluginhost process
Child process with PID 176142 exited with error code: 1
Waiting for plugin host process to exit
Plugin host process exited with code: 256
```
Can also submit this to the Bitwig team if you think the issue is on their end.

**Comment by mkruselj** [comment]
I think Bitwig is broken with Surge since 3.2.5 or something (or was it 3.2.4). Up to Bitwig team to fix IIRC.

**Comment by baconpaul** [comment]
Key board entry on all vstgui vst3s has been broken since bw325 on linux yes. Not just surge

We’ve been in touch with the bw team about this but let me ping them again today

**Comment by baconpaul** [comment]
I dropped them a note again. Lets see!
I'll leave this open but alas there's nothing we've been able to do in surge (or the vst3 stock plugs) yet.
Will update if/when we have resolution.

**Comment by jc-SpaceXp** [comment]
Thanks for the quick update!

**Comment by jc-SpaceXp** [comment]
> I think Bitwig is broken with Surge since 3.2.5 or something (or was it 3.2.4). Up to Bitwig team to fix IIRC. Can you try either of those two versions I mentioned and see if it works for you?

You are right! Downgraded to 3.2.4 and surge seems to be working as expected :)

**Comment by baconpaul** [comment]
Just some more on this: This is still the case up to BW 3.3. Turns out that if you run surge in non-sandboxed mode (so in the same process) the crash disappears, so i'm going to close this issue. I've shared the crash and the stack with the bitwig team and it's not just surge - it is any VST3SDK / VSTGUI plugin.

I think (think) either VSTGUI has a bad initialization or Bitwig is mis-handling some xkb resource when it forks the subprocess.
